File: src/metreos/util/Startable.java
/* $Id: Startable.java 21882 2006-04-11 18:26:09Z builder $
 *
 * Created by wert on Feb 10, 2006.
 * 
 * Copyright (c) 2007 Metreos, Inc. All rights reserved.
 */

package metreos.util;

/**
 * Description of Startable
 */
public interface Startable
{
	/**
	 * @throws Exception
	 */
	public void start() throws Exception;
	
	/**
	 * @throws Exception 
	 */
	public void stop() throws Exception;
	
	/**
	 * @return true if started.
	 */
	public boolean isStarted();
}

File: binding-csharp/runtime/src/main/csharp/Etch/Util/Startable.cs
// $Id$
// 
// Copyright 2007-2008 Cisco Systems Inc.
// 
// Licensed under the Apache License, Version 2.0 (the "License"); you may not
// use this file except in compliance with the License. You may obtain a copy
// of the License at
//  
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
// License for the specific language governing permissions and limitations
// under the License.

namespace Etch.Util
{
    /// <summary>
    /// Description of Startable
    /// </summary>
    public interface Startable
    {
        /// <summary>
        /// 
        /// </summary>
        /// Exception:
        ///     throws Exception
        void Start();

        /// Exception:
        ///     throws Exception
        void Stop();

        /// <summary>
        /// 
        /// </summary>
        /// <returns>true if started</returns>
        bool IsStarted();

        /// <summary>
        /// Check the startable is started
        /// </summary>
        void CheckIsStarted();
    
    }
}
